app/src/main/java/com/android/brewr/model/journey/JourneysRepositoryFirestore.kt

```diff
@@ -62,14 +62,93 @@ class JourneysRepositoryFirestore(
   }
 
   /**
-   * Retrieves all journeys from the Firestore database.
+   * Retrieves all journeys of current user from the Firestore database.
    *
    * @param onSuccess The callback to call with the list of journeys if the operation is successful.
    * @param onFailure The callback to call if the operation fails.
    */
-  override fun getJourneys(onSuccess: (List<Journey>) -> Unit, onFailure: (Exception) -> Unit) {
+  override fun getJourneysOfCurrentUser(
+      onSuccess: (List<Journey>) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    getJourneysOfTheUser(getCurrentUserUid(), onSuccess, onFailure)
+  }
+
+  /**
+   * Retrieves all journeys of users other than current user from the Firestore database.
+   *
+   * @param onSuccess The callback to call with the map of user uid and list of journeys if the
+   *   operation is successful.
+   * @param onFailure The callback to call if the operation fails.
+   */
+  override fun getJourneysOfAllOtherUsers(
+      onSuccess: (List<Pair<List<Journey>, String>>) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    // Fetch all user documents from the userPath collection
+    db.collection(userPath)
+        .get()
+        .addOnSuccessListener { querySnapshot ->
```

Comment:
I think here, we should use snapshotListener instead of get(). addOnSuccessListener because get.addOnSuccessListner doesn't make change for the offline mode. Can you check this part later when My PR is done? It will be merged into main very soon!